devlib/utils/android.py

```diff
@@ -177,7 +177,12 @@ def newline_separator(self):
     # versions of the ls tool in Android pre-v7.
     def _setup_ls(self):
         command = "shell '(ls -1); echo \"\n$?\"'"
-        output = adb_command(self.device, command, timeout=self.timeout)
+        try:
+            output = adb_command(self.device, command, timeout=self.timeout)
+        except subprocess.CalledProcessError as e:
+            raise HostError(
```

Comment:
Should be `TargetError` rather than `HostError`. The former is for errors that occur on target or with communication with the target; the latter is for issues with the host itself (missing dependencies, non-existing on-host paths etc).